lumicks/pylake/tests/test_imaging_confocal/test_kymo_slice_crop.py

```diff
@@ -0,0 +1,273 @@
+    kymo, ref = test_kymo
+    ref_pixels = ref.metadata.pixels_per_line
+    ref_lines = ref.metadata.lines_per_frame
+
+    scan_time = (ref.timestamps.dt * ref.infowave.samples_per_pixel * ref_pixels) * 1e-9
+    dead_time = (ref.timestamps.dt * ref.infowave.line_padding * 2) * 1e-9
+    line_time = scan_time + dead_time
+
+    deadtime_slice_offset = (1 + ref.infowave.line_padding) * ref.timestamps.dt * 1e-9
+
+    assert kymo.get_image("red").shape == (ref_pixels, ref_lines)
+    assert kymo.shape == (ref_pixels, ref_lines, 3)
+    np.testing.assert_allclose(kymo.get_image("red").data, ref.image[:, :, 0])
+
+    sliced = kymo[:]
+    assert sliced.get_image("red").shape == (ref_pixels, ref_lines)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :, 0])
+
+    # Anything will crop of the first frame
+    # need to make sure you're past the deadtime
+    # todo: for 2.0 define the start of the kymo as the start of the actual image, not deadtime
+    sliced = kymo[f"{deadtime_slice_offset}s":]
+    assert sliced.get_image("red").shape == (ref_pixels, ref_lines - 1)
+    assert sliced.shape == (ref_pixels, ref_lines - 1, 3)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, 1:, 0])
+
+    sliced = kymo["0s":]
+    assert sliced.get_image("red").shape == (ref_pixels, ref_lines)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :, 0])
+
+    factor = 2
+    sliced = kymo["0s":f"{factor * line_time}s"]
+    assert sliced.get_image("red").shape == (ref_pixels, factor)
+    assert sliced.shape == (ref_pixels, factor, 3)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :factor, 0])
+
+    sliced = kymo["0s":f"-{line_time * 0.6}s"]
+    assert sliced.get_image("red").shape == (ref_pixels, ref_lines - 1)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :-1, 0])
+
+    sliced = kymo["0s":f"-{2 * line_time}s"]
+    assert sliced.get_image("red").shape == (ref_pixels, ref_lines - 2)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, : (ref_lines - 2), 0])
+
+    # get a sliver of the next frame
+    # stop needs to be > halfway the deadtime between lines
+    sliced = kymo["0s":f"-{2 * line_time - deadtime_slice_offset}s"]
+    assert sliced.get_image("red").shape == (ref_pixels, ref_lines - 1)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :-1, 0])
+
+    # Two full frames
+    sliced = kymo["0s":f"{2 * line_time}s"]
+    assert sliced.get_image("red").shape == (ref_pixels, 2)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :2, 0])
+
+    # Two full frames plus a bit
+    sliced = kymo["0s":f"{2 * scan_time + 2 * dead_time + deadtime_slice_offset}s"]
+    assert sliced.get_image("red").shape == (ref_pixels, 3)
+    np.testing.assert_allclose(sliced.get_image("red").data, ref.image[:, :3, 0])
+
+    sliced = kymo["3.5s":"14s"]
```

Comment:
Why go back to seconds here rather than something expressed in line times and whatnot?